src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -138,6 +138,13 @@ def __init__(  # pylint: disable=too-many-arguments
         self._resampler_sender: Sender[Sample] | None = None
         self._resampler_task: asyncio.Task[None] | None = None
 
+        self.count_samples = 0
+        """The number of samples that have been received."""
+        self.wait_for_num_samples = 0
+        """The number of samples to wait for before
+        the wait_for_samples method triggers."""
+        self.wait_for_samples_event = asyncio.Event()
```

Comment:
These two **really** look like they should be *private*. Same about names (and missing types):

```suggestion
        self._expected_samples_count: int = 0
        """The number of samples to wait for before `wait_for_samples()` triggers."""
        self._wait_for_samples_event: asyncio.Event = asyncio.Event()
        """The event to signal `wait_for_samples()` that the wait is over."""
```